sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -912,4 +1046,26 @@ public static void ResetFeatureSwitch()
                 "AZURE_EXPERIMENTAL_ENABLE_ACTIVITY_SOURCE");
         }
     }
+#else
+#pragma warning disable SA1507 // File can not contain multiple types
+    /// <summary>
+    /// Until Activity Source is no longer considered experimental.
+    /// </summary>
+    internal static class ActivityExtensions
```

Comment:
Consider finding a different place for these methods so we can delete ActivityExtensions altogether